### PR TITLE
Add GitHub authentication

### PR DESCRIPTION
**Pull request by harshshahsumo**
This PR will help in making request to github after authentication. Related [issue](https://github.com/SumoLogic/sumologic-otel-collector-packaging/issues/79)

### CODE REVIEW

File: install-script/test/consts_common.go [hunk 1]
```diff
@@ -18,18 +18,36 @@ var (
 	latestAppVersion string
 )
 
+func authenticateGithub() string {
+	githubToken := os.Getenv("GH_CI_TOKEN")
+	if githubToken == "" {
+		fmt.Println("Error: GitHub token not found in environment variables")
+		os.Exit(1)
```
Comment (echlebek) [inline install-script/test/consts_common.go:25]: You can use log.Fatal here, FYI

File: .github/workflows/_reusable_build_package.yml [hunk 2]
```diff
@@ -66,6 +68,8 @@ jobs:
     outputs:
       package_path: ${{ steps.package.outputs.path }}
     steps:
+      - name: Debug GitHub Token
+        run: echo "Token is ${{ secrets.GH_CI_TOKEN }}"
```
Comment (echlebek) [inline .github/workflows/_reusable_build_package.yml:72]: This should be removed, and we should change the token afterwards

**Review by echlebek — CHANGES_REQUESTED** [review]

File: install-script/test/consts_common.go (hunk 1 above)
Comment (harshshahsumo) [reply install-script/test/consts_common.go]: Addressed

File: .github/workflows/_reusable_build_package.yml (hunk 2 above)
Comment (harshshahsumo) [reply .github/workflows/_reusable_build_package.yml]: Addressed

**Review by echlebek — APPROVED** [review]

**Pull request merged**